README.md

```diff
@@ -1,46 +1,103 @@
-
-- ReadString
-`string secret = SecretHub.Client.ReadString("path_to_secret");`
-
-- Exists
-`bool isSecret = SecretHub.Client.Exists("path_to_secret");`
-
-- Write
-`SecretHub.Client.Write("path_to_secret", "secret_value");`
+## Building from source 
+1. Execute `make nupkg` from the Makefile
+2. Go to your .NET project direcotry and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
 
-- Remove
-`SecretHub.Client.Remove("path_to_secret");`
-
-- Resolve
-`string secret = SecretHub.Client.Resolve("reference");`
-
-- ResolveEnv
-`Doctionary<string, string> resolvedEnv = SecretHub.Client.ResolveEnv(dictionaryToResolve);`
-
-## Example
-
-An example project is found in `azure-example`. There you can see the library in action.
+## How to call library functions
+Before doing any calls to the library, you need to create you SecretHub client. This is done in the following way:
+```csharp
+var client = new SecretHub.Client();
+``` 
+or 
+```csharp
+SecretHub.Client client = new SecretHub.Client();
+```
+
+After you have your client, you can perform on of the following functions:
+
+### `Read(string path)`
+Retrieve a secret, including all its metadata:
+```csharp
+SecretHub.SecretVersion secret = client.Read("path/to/secret");
+Console.WriteLine("The secret value is " + secret.Data);
+```
+`SecretHub.SecretVersion` object represents a version of a secret with sensitive data.
+
+### `ReadString(string path)`
+Retrieve a secret as a string:
+ ```csharp
+ string secret = client.Read("path/to/secret");
+ Console.WriteLine("The secret value is " + secret);
+ ```
+
+### `Exists(string path)`
+Check if a secret exists at `path`:
+```csharp
+bool isSecret = client.Exists("path/to/secret");
+if (isSecret) 
+{
+    Console.WriteLine("The secret exists.");
+} 
+else 
+{
+    Console.WriteLine("The secret does not exists.");
+}
+```
+
+### `Write(string path, string secret)`
+Write a secret containing the contents of `secret` at `path`:
+```csharp
+client.Write("path/to/secret", "secret_value");
+```
+
+### `Remove(string path)`
+Delete the secret found at `path`, if it exists:
+```csharp
+client.Remove("path/to/secret");
+if (!client.Exists("path/to/secret"))
+{
+    Console.WriteLine("Secret deleted successfully");
+}
+```
+
+### `Resolve(string ref)`
+Fetch the value of a secret from SecretHub, when the `ref` has the format `secrethub://<path>`, otherwise it returns `ref` unchanged:
+```csharp
+string resolvedRef = client.Resolve("secrethub://path/to/secret");
+Console.WriteLine("The secret value got from reference is " + resolvedRef);
+```
+
+### `ResolveEnv()`
+Take a map of system's environment variables and replaces the values of those which store references of secrets in SecretHub (`secrethub://<path>`) with the value of the respective secret. The other entries in the map remain untouched.
+```csharp
+Doctionary<string, string> resolvedEnv = client.ResolveEnv(dictionaryToResolve);
```

Comment:
I think an example showing an input map and the resulting map could be really helpful here.

For example (pseudo-code; I don't speak fluent C#):
```csharp
var dictionaryToResolve = Dictionary<string, string>{
    "reference_to_secret": "secrethub://path/to/secret",
    "not_a_reference": "some-other-value" 
}
var resolvedEnv = client.ResolveEnv(dictionaryToResolve);
// resolvedEnv = Dictionary<string, string>{
//    "reference_to_secret": "some-secret-value",
//    "not_a_reference": "some-other-value" 
// }

```

Oo, and I think there's a typo in `Dictionary`.

Reply:
I realised I made a mistake. `ResolveEnv` does not require any parameters. Therefore, we do not need the `dictionaryToResolve` example. I will think quickly of an example that suits it. If you have ideas, let me know

Reply:
You're totally right! In that case, we should still try to make it as clear as possible to the user what is happening.

What about the following? It breaks a little with the style of the other functions, but by being very verbose, we may be able to take away some questions the user might have.

````md
### `ResolveEnv()`
Load all of the process's environment variables in a dictionary and replace all values that are a reference to a secret  (`secrethub://<path>`) to the value as stored in SecretHub. The other values in the dictionary remain untouched.

For example, if the following two environment variables are set:
- `MY_SECRET=secrethub://path/to/secret`
- `OTHER_VARIABLE=some-other-value`

Then the following call to `ResolveEnv()`
```csharp
Dictionary<string, string> resolvedEnv = client.ResolveEnv();
```

would lead to the `resolvedEnv` containing the following contents:
```csharp
Dictionary<string, string> {
	{"MY_SECRET", "the value of the secret path/to/secret"},
	{"OTHER_VARIABLE": "some-other-value"}
};
```
````

Reply:
Sweet. Adjusted